lib/exec.cc

```diff
@@ -25,50 +53,18 @@ NAN_METHOD(WrappedRE2::Exec) {
 
 	vector<char> buffer;
 
-	char*  data;
-	size_t size, lastIndex = 0;
-	bool   isBuffer = false;
-
-	if (node::Buffer::HasInstance(info[0])) {
-		isBuffer = true;
-		size = node::Buffer::Length(info[0]);
-		if (re2->global) {
-			if (re2->lastIndex > size) {
-				re2->lastIndex = 0;
-				info.GetReturnValue().SetNull();
-				return;
-			}
-			lastIndex = re2->lastIndex;
-		}
-		data = node::Buffer::Data(info[0]);
-	} else {
-		if (re2->global && re2->lastIndex) {
-			String::Value s(info[0]->ToString());
-			if (re2->lastIndex > s.length()) {
-				re2->lastIndex = 0;
-				info.GetReturnValue().SetNull();
-				return;
-			}
-			Local<String> t(Nan::New(*s + re2->lastIndex).ToLocalChecked());
```

Comment:
I'm looking for a way to check in which cases this optimization can safely be restored, and to do so when possible.